youtube_dl/extractor/tvp.py

```diff
@@ -1,139 +1,247 @@
+            'title': 'Boso przez świat',
+            'id': '9329207',
+            'description': 'Docieramy do plemion w zapomnianych regionach naszej planety. Poznajemy ich kulturę, wierzenia i zwyczaje. Na ile są podobne do naszych? Wojciech Cejrowski jest naszym przewodnikiem po najbardziej dzikich zakątkach globu.',
         },
+        'playlist_count': 86,
     }]
 
-    def _real_extract(self, url):
-        video_id = self._match_id(url)
+    def _real_initialize(self):
+        self.api = TvpApi(self)
+
+    @staticmethod
+    def _format_formats(formats, video_id):
 
-        webpage = self._download_webpage(
-            'http://www.tvp.pl/sess/tvplayer.php?object_id=%s' % video_id, video_id)
-
-        title = self._search_regex(
-            r'name\s*:\s*([\'"])Title\1\s*,\s*value\s*:\s*\1(?P<title>.+?)\1',
-            webpage, 'title', group='title')
-        series_title = self._search_regex(
-            r'name\s*:\s*([\'"])SeriesTitle\1\s*,\s*value\s*:\s*\1(?P<series>.+?)\1',
-            webpage, 'series', group='series', default=None)
-        if series_title:
-            title = '%s, %s' % (series_title, title)
-
-        thumbnail = self._search_regex(
-            r"poster\s*:\s*'([^']+)'", webpage, 'thumbnail', default=None)
-
-        video_url = self._search_regex(
-            r'0:{src:([\'"])(?P<url>.*?)\1', webpage, 'formats', group='url', default=None)
-        if not video_url:
-            video_url = self._download_json(
-                'http://www.tvp.pl/pub/stat/videofileinfo?video_id=%s' % video_id,
-                video_id)['video_url']
-
-        ext = video_url.rsplit('.', 1)[-1]
-        if ext != 'ism/manifest':
-            if '/' in ext:
-                ext = 'mp4'
-            formats = [{
-                'format_id': 'direct',
-                'url': video_url,
-                'ext': ext,
-            }]
+        mime_ext = {
+            'video/x-ms-wmv': 'wmv',
+            'video/mp4': 'mp4'
+        }
+
+        viable_formats = []
+        for f in formats:
+            if f['mimeType'] in IGNORED_MIMETYPES:
+                continue
+
+            elif f['mimeType'].startswith('video/'):
+                viable_formats.append(
+                    {'url': f['url'],
+                     'ext': mime_ext.get(f['mimeType'], None),
+                     'vbr': f['totalBitrate']})
+
+        return viable_formats
+
+    def _get_video(self, context):
+        id = context['material_id']
+        if context['title_root']:
+            title = context['title_root']
+        elif not context['website_title']:
+            title = context['title']
         else:
-            m3u8_url = re.sub('([^/]*)\.ism/manifest', r'\1.ism/\1.m3u8', video_url)
-            formats = self._extract_m3u8_formats(m3u8_url, video_id, 'mp4')
+            title = ', '.join([context['website_title'], context['title']])
+        url = context['url']
+        description = context['description_root']
```

Comment:
API always provides that key, even if value is just an empty string, should I handle it nonetheless?